### PR TITLE
Issue #3406 - restore jetty websocket tests

### PR DESCRIPTION
**Pull request by lachlan-roberts**
#3406

### CODE REVIEW

File: jetty-websocket/javax-websocket-tests/src/test/java/org/eclipse/jetty/websocket/javax/tests/server/SessionTrackingTest.java [hunk 1]
```diff
@@ -18,30 +18,29 @@
...
-import java.io.IOException;
-import java.util.ArrayList;
-import java.util.Arrays;
-import java.util.Collection;
-import java.util.List;
-import java.util.concurrent.TimeUnit;
-
 import static org.junit.jupiter.api.Assertions.assertNotNull;
 
-@Disabled("Replaced with ServerCloseTest.testOpenSessionCleanup()") // TODO: Remove once ServerCloseTest is fixed
```
Comment (lachlan-roberts) [inline jetty-websocket/javax-websocket-tests/src/test/java/org/eclipse/jetty/websocket/javax/tests/server/SessionTrackingTest.java]: @joakime should i remove this test now that `ServerCloseTest.testOpenSessionCleanup()` is working

Comment (lachlan-roberts) [reply jetty-websocket/javax-websocket-tests/src/test/java/org/eclipse/jetty/websocket/javax/tests/server/SessionTrackingTest.java]: this test was failing due to a typo in `SessionTracker` which `ServerCloseTest.testOpenSessionCleanup()` did not pick up, so probably worth keeping this test

File: jetty-websocket/jetty-websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java [hunk 2]
```diff
@@ -61,23 +63,17 @@
...
      * Instantiate a WebSocketClient with defaults
      */
     public WebSocketClient()
     {
-        this(new WebSocketCoreClient());
-        this.coreClient.getHttpClient().setName("Jetty-WebSocketClient@" + hashCode());
-        // We created WebSocketCoreClient, let lifecycle be managed by us
-        addManaged(coreClient);
+        this(null);
+        coreClient.getHttpClient().setName("Jetty-WebSocketClient@" + hashCode());
```
Comment (gregw) [inline jetty-websocket/jetty-websocket-client/src/main/java/org/eclipse/jetty/websocket/client/WebSocketClient.java]: This should be in the main constructor and done after creating the coreClient if the passed in httpClient is null

**Comment by lachlan-roberts** [comment]
@gregw does this need another review or can you merge

**Pull request merged**